code/script.js

```diff
@@ -244,100 +250,151 @@ const generateBoard = () => {
+      )
     } else {
-      // alert popup that says something like: "NO, the person doesnt have yellow hair! Remove all persons with yellow hair"
+      alert(
+        `No, the person is not a ${currentQuestion.attribute}! Remove all persons that is a ${currentQuestion.attribute}`
+      )
     }
-  }
-
-  // filter to keep or remove based on the keep variable.
-  /* charactersInPlay = charactersInPlay.filter((person) => person[attribute] === value)
-    or 
-    charactersInPlay = charactersInPlay.filter((person) => person[attribute] !== value) */
+  } else if (group === 'hair color') {
+    if (keep) {
+      alert(
+        `Yes, the person has ${currentQuestion.value} hair! Keep all that has ${currentQuestion.value} hair`
+      )
+    } else {
+      alert(
+        `No, the person doesn't have ${currentQuestion.value} hair! Remove all persons with ${currentQuestion.value} hair`
+      )
+    }
+  } else if (group === 'eye color') {
+    if (keep) {
+      alert(
+        `Yes, the person has ${currentQuestion.value} eyes! Keep all that has ${currentQuestion.value} eyes`
+      )
+      
+    } else {
+      alert(
+        `No, the person doesn't have ${currentQuestion.value} eyes! Remove all persons with ${currentQuestion.value} eyes`
+      )
+    }
+  } 
 
-  // Invoke a function to redraw the board with the remaining people.
+  if (keep) {
+  charactersInPlay = charactersInPlay.filter((person) => person[currentQuestion.attribute] === currentQuestion.value)
+ } else {
+    charactersInPlay = charactersInPlay.filter((person) => person[currentQuestion.attribute] !== currentQuestion.value)
+  }
+  generateBoard()
 }
 
-// when clicking guess, the player first have to confirm that they want to make a guess.
+  
+
+// Guess section
 const guess = (suspect) => {
-  // store the interaction from the player in a variable.
-  // remember the confirm() ?
-  // If the player wants to guess, invoke the checkMyGuess function.
+  let confirmAnswer = confirm('Are you sure?')
+  if (confirmAnswer === true) {
+    checkMyGuess(suspect)
+  } else {
+    alert(
+      `Sorry...! Try again`
+    )
+  }
 }
 
-// If you confirm, this function is invoked
 const checkMyGuess = (suspect) => {
-  // 1. Check if the suspect is the same as the secret person's name
-  // 2. Set a Message to show in the win or lose section accordingly
-  // 3. Show the win or lose section
-  // 4. Hide the game board
+  
+  if (suspect === secret.name) {
+    winOrLoseText.innerHTML = `Yay! That's correct, ${suspect} it is!`
+
+  } else {
+    winOrLoseText.innerHTML = `Sorry, ${suspect} is wrong..`
+  }
+  board.innerHTML=""
+  winOrLose.classList.add("show")
 }
 
-// Invokes the start function when website is loaded
+// Start function
 start()
 
-// All the event listeners
+
+// Event listeners
+playAgian.addEventListener("click", () => {winOrLose.classList.remove("show"); start()})
```

Comment:
This part here is part of the red level requirements as I recall it, but since you’ve done the code I’ll look at this as well. 😊

Generally you can use this syntax (i.e. writing everything on one line) if there’s only one line of code. Your function here though contains two lines of code. Also, now you used a semicolon which you’re not using in the rest of the file. In the spirit of being consistent, I suggest you remove the semicolon and that you instead write the code (if you want to write it here) like this: 

```
playAgian.addEventListener("click", () => {
winOrLose.classList.remove("show") 
start()
})
```